src/img2pdf.py

```diff
@@ -212,11 +213,26 @@ def tostring(self):
         result += b"%%EOF\n"
         return result
 
-def convert(images, dpi=None, pagesize=(None, None), title=None, author=None,
+def convert(images, dpi=None, pagesize=(None, None, None), title=None, author=None,
             creator=None, producer=None, creationdate=None, moddate=None,
             subject=None, keywords=None, colorspace=None, nodate=False,
             verbose=False):
 
+    pagesize_options = {
+        'ignore'  : ['.*\!.*', False],
+        'shrink'  : ['.*\>.*', False],
+        'enlarge' : ['.*\<.*', False],
+        'fill'    : ['.*\^.*', False],
+        'percent' : ['.*\%.*', False],
+        'count'   : ['.*\@.*', False],
+    }
+
+    if pagesize[2]:
+        for key, value in pagesize_options.items():
+            pattern = re.compile(value[0])
```

Comment:
the compilation of the regex should happen above so that there are not multiple compilations